modules/loader/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/config/Config.scala

```diff
@@ -70,7 +71,13 @@ object Config {
   final case class StatsD(hostname: String, port: Int, tags: Map[String, String], prefix: Option[String])
   final case class Stdout(prefix: Option[String])
   final case class Webhook(endpoint: Uri, tags: Map[String, String])
-  final case class Folders(period: FiniteDuration, staging: S3.Folder, since: Option[FiniteDuration], transformerOutput: S3.Folder, until: Option[FiniteDuration], failBeforeAlarm: Option[Int])
+  final case class Folders(period: FiniteDuration,
+                           staging: S3.Folder,
+                           since: Option[FiniteDuration],
+                           transformerOutput: S3.Folder,
+                           until: Option[FiniteDuration],
+                           failBeforeAlarm: Option[Int],
+                           appendStagingPath: Option[Boolean])
```

Comment:
Actually, I've implemented it like as you suggested initially but I found out it leads to a problem. 

When this field is added to `application.conf`, if `monitoring.folders` section isn't provided in the config, it throws parsing error because it expects other non-optional fields of `folders` section to be in the config as well. Therefore, we can't provide default for `appendStagingPath` only. If we want to include it to `application.conf`, we need to provide defaults for `period`, `staging` and `transformerOutput` as well. Since there is no meaningful defaults for this fields, I've made `appendStagingPath`'s type `Option` instead.

Please let me know if you have an idea to make it work.

Reply:
Ah yes, I remember that problem from other places, and I don't know a nice solution.

One day I would like to find a neat solution and use it everywhere, but we don't need to solve it now.